### ISSUE TITLE
Appcenter is unable to find Android build

### ISSUE THREAD

**Issue by sawankumarbundelkhandi** [microsoft/appcenter#1520]
The app center is unable to find artifact after a successful build.

**Describe the solution you'd like**
Fix - https://github.com/microsoft/appcenter/pull/1519

**Additional context**

Below is the script that is executed internally to find the build and the error I was getting.

Task         : Shell Script
Description  : Run a shell script using bash
Version      : 2.1.3
Author       : Microsoft Corporation
Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613738)

[command]/bin/bash /Users/runner/runners/2.163.1/scripts/android-postprocess.sh /Users/runner/runners/2.163.1/work/1/s/android/app/app/build/outputs/apk
Removing all ABI or density dependent APKs…
find: /Users/runner/runners/2.163.1/work/1/s/android/app/app/build/outputs/apk: No such file or directory
find: /Users/runner/runners/2.163.1/work/1/s/android/app/app/build/outputs/apk: No such file or directory
find: /Users/runner/runners/2.163.1/work/1/s/android/app/app/build/outputs/apk: No such file or directory
Found        0 APK file(s)
Found        0 unaligned APK file(s)
find: /Users/runner/runners/2.163.1/work/1/s/android/app/app/build/outputs/apk: No such file or directory
/Users/runner/runners/2.163.1/scripts/android-postprocess.sh: line 36: pushd: /Users/runner/runners/2.163.1/work/1/s/android/app/app/build/outputs/apk/..: No such file or directory
/Users/runner/runners/2.163.1/scripts/android-postprocess.sh: line 43: popd: directory stack empty
##[error]/bin/bash failed with return code: 1
##[error]Bash failed with error: /bin/bash failed with return code: 1

**Comment by baotoan1905** [comment]
Same issue, any idea ?

**Comment by sawankumarbundelkhandi** [comment]
@baotoan1905 check this pr - https://github.com/microsoft/appcenter/pull/1519/files

**Comment by derpixeldan** [comment]
Thanks for noting, merged the PR.

**Comment by makassigithub** [comment]
this issue should be reopened. I am still experiencing the same issue with react-native 0.61.5

**Comment by saileshkasaju** [comment]
still getting this issue. Please reopen the issue. this pr reverted the change: [Revert "Updated path so that appcenter can find the apk"](https://github.com/microsoft/appcenter/commit/264a9c4555c665059516544e7c6ae9bd90f1e077)

```
find: /Users/runner/runners/2.164.8/work/1/s/android/app/app/build/outputs/apk: No such file or directory
Found        0 APK file(s)
```

**Comment by Prosquid1** [comment]
Ensure you do not have gradle.bat and a gradle folder in your app directory.
This fixed my issue.

**Comment by dusanjovanov** [comment]
> Ensure you do not have gradle.bat and a gradle folder in your app directory.
> This fixed my issue.

When I do this I get this error:
```
==============================================================================
Task         : Gradle
Description  : Build using a Gradle wrapper script
Version      : 1.128.0
Author       : Microsoft Corporation
Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613720)
==============================================================================
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2373) Warning: Use Cipheriv for counter mode of aes-256-ctr
##[error]Error: Not found wrapperScript: /Users/runner/runners/2.165.2/work/1/s/mobile/android/app/gradlew
##[section]Finishing: Gradle Task
##[section]Starting: Post Build Script
```

**Comment by seanoh1989** [comment]
Any update? I'm getting the same issue.

**Comment by sawankumarbundelkhandi** [comment]
This issue was related to flutter and the script change I gave was for the flutter. 
I think we should open a new issue for react-native.

**Comment by ptsiogas** [comment]
I am getting the same issue. My project is native android.

**Comment by Prosquid1** [comment]
@ptsiogas I think the problem is your project has unrelated grade files in the 'app' directory.

If you have unrelated grade files in the 'app' directory, your app will proceed to build an app using the 'app' directory instead of the 'android' directory and you will end up with an MSCenter output apk with path: 'android/**app**/**app**/build/outputs/apk'

As opposed to 'android/**app**/build/outputs/apk'

**Comment by ptsiogas** [comment]
> @ptsiogas I think the problem is your project has unrelated grade files in the 'app' directory.
> 
> If you have unrelated grade files in the 'app' directory, your app will proceed to build an app using the 'app' directory instead of the 'android' directory and you will end up with an MSCenter output apk with path: 'android/**app**/**app**/build/outputs/apk'
> 
> As opposed to 'android/**app**/build/outputs/apk'

How can I fix that?

**Comment by Prosquid1** [comment]
Delete all **unrelated** files in your **app** directory, 'gradle.bat', gradle wrapper properties, 
You can start a new Android project and compare your app directory with what you currently have.

**Comment by ptsiogas** [comment]
> Delete all **unrelated** files in your **app** directory, 'gradle.bat', gradle wrapper properties,
> You can start a new Android project and compare your app directory with what you currently have.

Thanks. I will try those steps and let you know.

**Comment by ptsiogas** [comment]
> > Delete all **unrelated** files in your **app** directory, 'gradle.bat', gradle wrapper properties,
> > You can start a new Android project and compare your app directory with what you currently have.
> 
> Thanks. I will try those steps and let you know.

It worked like a charm! Very useful tip!

**Comment by Spelliar** [comment]
Same issue. Android native. My app folder is clear.
<img width="291" alt="Снимок экрана 2020-08-05 в 09 42 47" src="https://user-images.githubusercontent.com/3800323/89380381-0f9ada80-d700-11ea-84cc-4bd086a1add7.png">
May be anyone knows other decisions?

**Comment by johnclarete** [comment]
> Same issue. Android native. My app folder is clean.
> <img alt="Снимок экрана 2020-08-05 в 09 42 47" width="291" src="https://user-images.githubusercontent.com/3800323/89380381-0f9ada80-d700-11ea-84cc-4bd086a1add7.png">
> May be anyone knows other decisions?

@Spelliar did you get any progress on this am also getting the same issue as yourself.

**Comment by affantechcarrot** [comment]
@makassigithub How did you resolve this.i am facing similar issue in appcenter

**Comment by seanoh1989** [comment]
I was able to resolve this by removing duplicated gradle-wrapper. You only need one

**Comment by affantechcarrot** [comment]
@seanoh1989  I removed but i am getting this now 
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
##[error]Error: Not found wrapperScript: /Users/runner/work/1/s/android/app/gradlew
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
(node:2301) Warning: Use Cipheriv for counter mode of aes-256-ctr
##[section]Finishing: Gradle Task

**Comment by johnclarete** [comment]
@affantechcarrot I too was getting the same issue as you after removing the duplicate grade wrapper files in the app folder. Still would like to know why this isn't working but a workaround I found that was working was to add this into the module level build.gradle for the different build types like in the example below

release {
applicationVariants.all { variant ->
                variant.outputs.all {
                    def currentFile = new File(outputFileName)
                    def filename = currentFile.getName()
                    outputFileName = "../../../../../app/build/outputs/apk/${filename}"
                }
}

**Comment by affantechcarrot** [comment]
@johnclarete what i did. i just click on save and build  and the appcenter created the build.

**Comment by stanhughes** [comment]
> @ptsiogas I think the problem is your project has unrelated gradle files in the 'app' directory.
> 
> If you have unrelated gradle files in the 'app' directory, your app will proceed to build an app using the 'app' directory instead of the 'android' directory and you will end up with an MSCenter output apk with path: 'android/**app**/**app**/build/outputs/apk'
> 
> As opposed to 'android/**app**/build/outputs/apk'

Thank you so much and thank you for having the patience to explain it more than one. This fixed the problem for me. However, I also had to delete the original AppCenter build configuration because the scripts files where based off the bad Grade definitions and simply updating the branch didn't update the script files. After creating a new configuration definition, everything built fine for my Android AppCenter build.

**Comment by SameeraMadushan** [comment]
I had the same issue cause of generating multiple .apk with `enableSeparateBuildPerCPUArchitecture = true`
Solution:
https://github.com/microsoft/appcenter/issues/501#issuecomment-547142908

**Comment by TylerEikenberg** [comment]
@dipree 
@dusanjovanov @SameeraMadushan @stanhughes  @affantechcarrot @johnclarete

Hi, was anyone here actually able to solve this issue? I'm having the exact problem and App Center support doesn't seem to be able to help me. My android app builds fine locally but in App Center im getting this error. 
```
##[section]Starting: Gradle Task
==============================================================================
Task         : Gradle
Description  : Build using a Gradle wrapper script
Version      : 1.128.0
Author       : Microsoft Corporation
Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613720)
==============================================================================
##[error]Error: Not found wrapperScript: /Users/runner/work/1/s/android/app/gradlew
(node:2691) Warning: Use Cipheriv for counter mode of aes-256-ctr
```

Here is my file structure
![Screen Shot 2020-12-03 at 5 14 37 PM](https://user-images.githubusercontent.com/55251251/101094839-070dc000-358b-11eb-8b6d-23617e775f30.png)

Is anyone able to see what is wrong? I'd really appreciate the help. Please let me know if I can provide any further context.

**Comment by anhdo9797** [comment]
@stanhughes  Can you give more detailed instructions?

**Comment by anhdo9797** [comment]
> @dipree
> @dusanjovanov @SameeraMadushan @stanhughes @affantechcarrot @johnclarete
> 
> Hi, was anyone here actually able to solve this issue? I'm having the exact problem and App Center support doesn't seem to be able to help me. My android app builds fine locally but in App Center im getting this error.
> 
> ```
> ##[section]Starting: Gradle Task
> ==============================================================================
> Task         : Gradle
> Description  : Build using a Gradle wrapper script
> Version      : 1.128.0
> Author       : Microsoft Corporation
> Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613720)
> ==============================================================================
> ##[error]Error: Not found wrapperScript: /Users/runner/work/1/s/android/app/gradlew
> (node:2691) Warning: Use Cipheriv for counter mode of aes-256-ctr
> ```
> 
> Here is my file structure
> ![Screen Shot 2020-12-03 at 5 14 37 PM](https://user-images.githubusercontent.com/55251251/101094839-070dc000-358b-11eb-8b6d-23617e775f30.png)
> 
> Is anyone able to see what is wrong? I'd really appreciate the help. Please let me know if I can provide any further context.

I am getting the same issue

**Comment by TylerEikenberg** [comment]
@anhdo9797
I just solved this for myself. 

My issue was that the build branch ( the branch I configured App Center to build ) had `gradle` files in the `/android/app` directory. I had deleted these files in the brach I was attempting to merge into my build branch with a PR but it wasn't working until I directly deleted and committed the deletion on the build branch itself. Then on App Center I opened the config and chose `Save And Build` again.

 I can try to clarify if this isn't clear.

**Comment by anhdo9797** [comment]
> @anhdo9797
> I just solved this for myself.
> 
> My issue was that the build branch ( the branch I configured App Center to build ) had `gradle` files in the `/android/app` directory. I had deleted these files in the brach I was attempting to merge into my build branch with a PR but it wasn't working until I directly deleted and committed the deletion on the build branch itself. Then on App Center I opened the config and chose `Save And Build` again.
> 
> I can try to clarify if this isn't clear.


Can you give me see your file structure ? 

I deleted `gradle` in android and recived message 
![image](https://user-images.githubusercontent.com/60549336/101566672-f41f4380-3a01-11eb-8723-ca4208b0d30a.png)

**Comment by TylerEikenberg** [comment]
> > @anhdo9797
> > I just solved this for myself.
> > My issue was that the build branch ( the branch I configured App Center to build ) had `gradle` files in the `/android/app` directory. I had deleted these files in the brach I was attempting to merge into my build branch with a PR but it wasn't working until I directly deleted and committed the deletion on the build branch itself. Then on App Center I opened the config and chose `Save And Build` again.
> > I can try to clarify if this isn't clear.
> 
> Can you give me see your file structure ?
> 
> I deleted `gradle` in android and recived message
> ![image](https://user-images.githubusercontent.com/60549336/101566672-f41f4380-3a01-11eb-8723-ca4208b0d30a.png)

@anhdo9797 my file structure is posted above. It would actually be helpful to see your file structure if you dont mind?

**Comment by anhdo9797** [comment]
> > > @anhdo9797
> > > I just solved this for myself.
> > > My issue was that the build branch ( the branch I configured App Center to build ) had `gradle` files in the `/android/app` directory. I had deleted these files in the brach I was attempting to merge into my build branch with a PR but it wasn't working until I directly deleted and committed the deletion on the build branch itself. Then on App Center I opened the config and chose `Save And Build` again.
> > > I can try to clarify if this isn't clear.
> > 
> > 
> > Can you give me see your file structure ?
> > I deleted `gradle` in android and recived message
> > ![image](https://user-images.githubusercontent.com/60549336/101566672-f41f4380-3a01-11eb-8723-ca4208b0d30a.png)
> 
> @anhdo9797 my file structure is posted above. It would actually be helpful to see your file structure if you dont mind?

This is my files structure, I has restored back to the original. Can you help me ?
![image](https://user-images.githubusercontent.com/60549336/101709959-204ec900-3ac3-11eb-8d51-b5fad5acb941.png)

**Comment by TylerEikenberg** [comment]
@anhdo9797 This all looks fine. Do you also have a `gradlew`, `gradle.bat`, and `settings.gradle` in your `/android` directory?

**Comment by anhdo9797** [comment]
> @anhdo9797 This all looks fine. Do you also have a `gradlew`, `gradle.bat`, and `settings.gradle` in your `/android` directory?

My full files structure 
![image](https://user-images.githubusercontent.com/60549336/101789478-ee297f80-3b33-11eb-97e7-66771f3fe3ec.png)
****

**Comment by TylerEikenberg** [comment]
@anhdo9797 hmm, nothing here is immediately indicating a problem to me. Can you describe what is it you're trying to do, and what error you're getting?

**Comment by anhdo9797** [comment]
> @anhdo9797 hmm, nothing here is immediately indicating a problem to me. Can you describe what is it you're trying to do, and what error you're getting?
 I was like this: 
1. config app center
![image](https://user-images.githubusercontent.com/60549336/101801740-3c914b00-3b41-11eb-9ba6-b70d379909c1.png)
2. my structure: 
![image](https://user-images.githubusercontent.com/60549336/101802059-98f46a80-3b41-11eb-9b49-4c09a4d94355.png)
3. My `app/build.gradle`: https://gitlab.com/anhdo9797/test/-/blob/master/build.gradle

=> I recived error from app center 
![image](https://user-images.githubusercontent.com/60549336/101802606-364f9e80-3b42-11eb-8ae5-5900ca476e03.png)

**Comment by AuriR** [comment]
> @affantechcarrot I too was getting the same issue as you after removing the duplicate grade wrapper files in the app folder. Still would like to know why this isn't working but a workaround I found that was working was to add this into the module level build.gradle for the different build types like in the example below
> 
> release {
> applicationVariants.all { variant ->
> variant.outputs.all {
> def currentFile = new File(outputFileName)
> def filename = currentFile.getName()
> outputFileName = "../../../../../app/build/outputs/apk/${filename}"
> }
> }

Following up on @johnclarete idea above - it didn't work the first time for an Android native build, but I *did* get it working.

Steps:

1. Modify the build.gradle file in your Application folder as follows:

Find the android { part of the file and paste in the **buildTypes** part:

```
android {
    compileSdkVersion 30

	**buildTypes {
		appcenter {
			applicationVariants.all { variant ->
				variant.outputs.all {
					def currentFile = new File(outputFileName)
					def filename = currentFile.getName()
					outputFileName = "../../../../../Application/Application/build/outputs/apk/${filename}"
				}
			}
		}
	}**
```

2. In AppCenter, you should now have an "appcenter" Build Variant option.

![image](https://user-images.githubusercontent.com/3044108/101952591-e36bfa80-3bc6-11eb-8ccc-bb739e746f2d.png)

3. Disable Build Android App Bundle.

That worked for me. I hope that helps others.

**Comment by anhdo9797** [comment]
> 1. Disable Build Android App Bundle.

Can you help me how to `Disable Build Android App Bundle.` ?

**Comment by anhdo9797** [comment]
> 1. In AppCenter, you should now have an "appcenter

![image](https://user-images.githubusercontent.com/60549336/101975988-9b56e180-3c73-11eb-94f1-b870db4d9b3b.png)
Can you help me how to select Application ?
I only item project => pack.json

**Comment by anhdo9797** [comment]
> > @affantechcarrot I too was getting the same issue as you after removing the duplicate grade wrapper files in the app folder. Still would like to know why this isn't working but a workaround I found that was working was to add this into the module level build.gradle for the different build types like in the example below
> > release {
> > applicationVariants.all { variant ->
> > variant.outputs.all {
> > def currentFile = new File(outputFileName)
> > def filename = currentFile.getName()
> > outputFileName = "../../../../../app/build/outputs/apk/${filename}"
> > }
> > }
> 
> Following up on @johnclarete idea above - it didn't work the first time for an Android native build, but I _did_ get it working.
> 
> Steps:
> 
> 1. Modify the build.gradle file in your Application folder as follows:
> 
> Find the android { part of the file and paste in the **buildTypes** part:
> 
> ```
> android {
>     compileSdkVersion 30
> 
> 	**buildTypes {
> 		appcenter {
> 			applicationVariants.all { variant ->
> 				variant.outputs.all {
> 					def currentFile = new File(outputFileName)
> 					def filename = currentFile.getName()
> 					outputFileName = "../../../../../Application/Application/build/outputs/apk/${filename}"
> 				}
> 			}
> 		}
> 	}**
> ```
> 
> 1. In AppCenter, you should now have an "appcenter" Build Variant option.
> 
> ![image](https://user-images.githubusercontent.com/3044108/101952591-e36bfa80-3bc6-11eb-8ccc-bb739e746f2d.png)
> 
> 1. Disable Build Android App Bundle.
> 
> That worked for me. I hope that helps others.

My step:
1. add buildTypes:{
...
buildTypes:{
...
 appcenter {
			applicationVariants.all { variant ->
				variant.outputs.all {
					def currentFile = new File(outputFileName)
					def filename = currentFile.getName()
					outputFileName = "../../../../../Application/Application/build/outputs/apk/${filename}"
				}
			}
		}
}
}
2. config build 
![image](https://user-images.githubusercontent.com/60549336/101976705-82056380-3c7a-11eb-8a08-9879f85e2dcd.png)
=> error
![image](https://user-images.githubusercontent.com/60549336/101976723-947f9d00-3c7a-11eb-9c01-3f84f929c2c5.png)
[1_Build (1).txt](https://github.com/microsoft/appcenter/files/5682565/1_Build.1.txt)

Please help me !

**Comment by AuriR** [comment]
Hey there – if you send me your logs I’m happy to take a look. Make sure your app builds locally, otherwise AppCenter won’t be able to build it, either.

The toggle for Building App Bundle is in your Build Settings:

[cid:image003.png@01D6D065.B3AE30A0]

If you can confirm it builds, I’ll try to help further by looking at the logs.

Thanks and Best,

-Auri

--
Never confuse motion with action.
www.auri.net<http://www.auri.net> / @Auri on Twitter

**Comment by AuriR** [comment]
@anhdo9797 I saw your log file link. It looks like you have a React Native app, which I'm not familiar with.

However, it looks like you may have made the change in the wrong build.gradle file. I had to make sure I made the change to the one in my Application folder, where my Android source code was. I imagine you may need to do something similar with React Native, although I don't know where to tell you to look :)

**Comment by anhdo9797** [comment]
> Hey there – if you send me your logs I’m happy to take a look. Make sure your app builds locally, otherwise AppCenter won’t be able to build it, either. The toggle for Building App Bundle is in your Build Settings: [cid:image003.png@01D6D065.B3AE30A0] If you can confirm it builds, I’ll try to help further by looking at the logs. Thanks and Best,
> […](#)
> -Auri
> -- Never confuse motion with action. [www.auri.net<http://www.auri.net](http://www.auri.net<http://www.auri.net)> / @auri on Twitter From: Đỗ Công Phước Anh<mailto:notifications@github.com> Sent: Saturday, December 12, 2020 1:05 AM To: microsoft/appcenter<mailto:appcenter@noreply.github.com> Cc: Auri A. Rahimzadeh<mailto:aurir@msn.com>; Mention<mailto:mention@noreply.github.com> Subject: Re: [microsoft/appcenter] Appcenter is unable to find Android build (#1520) @affantechcarrot<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Faffantechcarrot&data=04%7C01%7C%7C24d53554f650431cf6c508d89e63f8f7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637433499491653779%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=UKFD6%2FTe5C6UYoJ6YroGh%2FQ3CcPYVE1C7ZVQ5R6jN3U%3D&reserved=0> I too was getting the same issue as you after removing the duplicate grade wrapper files in the app folder. Still would like to know why this isn't working but a workaround I found that was working was to add this into the module level build.gradle for the different build types like in the example below release { applicationVariants.all { variant -> variant.outputs.all { def currentFile = new File(outputFileName) def filename = currentFile.getName() outputFileName = "../../../../../app/build/outputs/apk/${filename}" } } Following up on @johnclarete<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjohnclarete&data=04%7C01%7C%7C24d53554f650431cf6c508d89e63f8f7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637433499491663775%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=NvhzeQXhnMphuug7UprqaAh9w9HHbX0AHYkj0E%2FpWmQ%3D&reserved=0> idea above - it didn't work the first time for an Android native build, but I did get it working. Steps: 1. Modify the build.gradle file in your Application folder as follows: Find the android { part of the file and paste in the buildTypes part: android { compileSdkVersion 30 **buildTypes { appcenter { applicationVariants.all { variant -> variant.outputs.all { def currentFile = new File(outputFileName) def filename = currentFile.getName() outputFileName = "../../../../../Application/Application/build/outputs/apk/${filename}" } } } }** 1. In AppCenter, you should now have an "appcenter" Build Variant option. [image]<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F3044108%2F101952591-e36bfa80-3bc6-11eb-8ccc-bb739e746f2d.png&data=04%7C01%7C%7C24d53554f650431cf6c508d89e63f8f7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637433499491663775%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=RqB4nohNaX8wAntzrCLzyuTz%2Fmk2rvk2y5IQTb7OiAo%3D&reserved=0> 1. Disable Build Android App Bundle. That worked for me. I hope that helps others. My step: 1. add buildTypes:{ ... buildTypes:{ ... appcenter { applicationVariants.all { variant -> variant.outputs.all { def currentFile = new File(outputFileName) def filename = currentFile.getName() outputFileName = "../../../../../Application/Application/build/outputs/apk/${filename}" } } } } } 2. config build [image]<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F60549336%2F101976705-82056380-3c7a-11eb-8a08-9879f85e2dcd.png&data=04%7C01%7C%7C24d53554f650431cf6c508d89e63f8f7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637433499491673767%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=D1EvAmmvEE8efnpnFuGCBZyog1XJU6J20OqWlpYkAd8%3D&reserved=0> => error [image]<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F60549336%2F101976723-947f9d00-3c7a-11eb-9c01-3f84f929c2c5.png&data=04%7C01%7C%7C24d53554f650431cf6c508d89e63f8f7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637433499491673767%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=agHn4qErbNQmnpDCZYh2PlyzXscHASx%2BFHRMPiZ7axw%3D&reserved=0> 1_Build (1).txt<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fappcenter%2Ffiles%2F5682565%2F1_Build.1.txt&data=04%7C01%7C%7C24d53554f650431cf6c508d89e63f8f7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637433499491683771%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=Q%2BS7Px5u67VrqI9pFQ2n8kJOw1eWEnQ2xmCsrTzR6vU%3D&reserved=0> Please help me ! — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fappcenter%2Fissues%2F1520%23issuecomment-743710216&data=04%7C01%7C%7C24d53554f650431cf6c508d89e63f8f7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637433499491683771%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=B0cT1ZKWbqBM5uSDeNlDg97YgMTpdmNJxqSZwItb1SY%3D&reserved=0>, or unsubscribe<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAXHGDDSLHNWCV3LZRXS7ELSUMB3ZANCNFSM4J54UPXQ&data=04%7C01%7C%7C24d53554f650431cf6c508d89e63f8f7%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637433499491683771%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=akmTaymVNPzVDkOmRRdtb0lWPvzVOO6z%2FXRhqiuj9Ic%3D&reserved=0>.

hi bros,  this is  my logs 

[1_Build.1.txt](https://github.com/microsoft/appcenter/files/5683294/1_Build.1.txt)

**Comment by gbadwaik2706** [comment]
In your csproj of Android, check OutputPath Variable is not set to "." (dot), if yes then change it as below:

<OutputPath>bin\Release</OutputPath>

**Comment by anhdo9797** [comment]
> In your csproj of Android, check OutputPath Variable is not set to "." (dot), if yes then change it as below:
> 
> bin\Release
> 
> Example
> < OutputPath> bin\Release </OutputPath>


Can you tell me more clearly, I don't understand your answer yet

**Comment by gbadwaik2706** [comment]
@anhdo9797 

You need to set OutputPath of your project configuration as shown in image attached.

![image](https://user-images.githubusercontent.com/4394119/102865744-3b72ee80-445c-11eb-9deb-ab2957e2cb9f.png)

**Comment by anhdo9797** [comment]
@gbadwaik2706  Sorry but The file could not be found, it could be added `OutputPath`
can you guide me where to configure it ?

**Comment by gbadwaik2706** [comment]
<img width="401" alt="Screenshot 2020-12-22 at 2 04 57 PM" src="https://user-images.githubusercontent.com/4394119/102867402-b937f980-445e-11eb-83c4-4b3793130e72.png">

You need to open the *.csproj file in TextEditor

**Comment by anhdo9797** [comment]
> <img alt="Screenshot 2020-12-22 at 2 04 57 PM" width="401" src="https://user-images.githubusercontent.com/4394119/102867402-b937f980-445e-11eb-83c4-4b3793130e72.png">
> 
> You need to open the *.csproj file in TextEditor

Sorry but I can't find in my project `react-native`
![image](https://user-images.githubusercontent.com/60549336/102868577-ebe8ee80-446c-11eb-8340-1510d11364ce.png)

**Comment by muneneevans** [comment]
In my case i had enabled building multiple APKs, but had disabled building a universal APK.

```
splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk true  // If true, also generate a universal APK
            include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
        }
    }
```
[Appcenter docs](https://docs.microsoft.com/en-us/appcenter/build/react-native/android/#63-building-multiple-apks)
